### PR TITLE
NIAD-143: store recep in inbound db

### PR DESCRIPTION
**Pull request by bartek-sarul**

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundState.java [hunk 1]
```diff
@@ -36,17 +38,31 @@ public static InboundState fromInterchange(Interchange interchange) {
...
             .setReceiveMessageSequence(messageHeader.getSequenceNumber())
             .setTransactionNumber(referenceTransactionNumber.getTransactionNumber())
             .setTransactionType(referenceTransactionType.getTransactionType())
-            .setTranslationTimestamp(interchangeHeader.getTranslationTime());
+            .setTranslationTimestamp(translationDateTime.getTimestamp());
+    }
+
+    public static InboundState fromRecep(Recep recep) {
+        var interchangeHeader = recep.getInterchangeHeader();
+        var dateTimePeriod = recep.getDateTimePeriod();
```
Comment (grzegorz-robak) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundState.java:59]: Would be better to name it with some context - what datetime is it?

**Review by grzegorz-robak — APPROVED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundState.java (hunk 1 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundState.java:59]: It's the dateTimePeriod segment

Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundState.java:59]: That I know - but what value is there? Transmission DT, Translation DT ?

**Review by mattd-kainos — APPROVED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundState.java (hunk 1 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/InboundState.java:59]: Line https://github.com/nhsconnect/integration-adaptor-nhais/pull/60/files/b41a077596d2f1cf5a16cdd5d8d761350f009498#diff-374424c2eec230a3d4af8ab11fe9704dR66 says where it's set

**Review by grzegorz-robak — APPROVED** [review]

**Pull request merged**